### ISSUE TITLE
Move mint.ini parsing into mint loader

### ISSUE THREAD

**Issue by mikrosk** [freemint/freemint#33]
Feature request / suggestion: to be able to offer the user list of possible kernels *and* give him the comfort of the boot menu in the same time, one major change is needed: move `mint.ini` parsing into the mint loader. So it would work like this:

1. mint loader is executed from `AUTO` and checks for presence of all possible `$SYSDIR`s in \\mint (i.e. `<boot drive>\mint`) -- this is possible thanks to the stable versioning scheme, `MAJOR-MINOR-PATCH_LEVEL` so the loader could nicely show "FreeMiNT 1.19.0", "FreeMiNT 1.19.DEV (123)" etc.
1. mint loader would use a shared `c:\mint\mint.ini` file which would actually *be* an [ini file](https://en.wikipedia.org/wiki/INI_file), i.e. with sections for each kernel
1. after the users chooses given kernel, he would get the list of options -- step by step, memory protection, debug level etc
1. after this step, the mint loader would execute the kernel for given CPU with arguments passed the same way how Aranym does it from its config file (i.e. as command line parameters to MINT.PRG)

Of course, there's nothing which actually forces are to remove `mint.ini` parsing (or even the boot menu) from MINT.PRG (keeping `$SYSDIR/mint.ini` still usable) but perhaps the functionality shouldn't be duplicated.

**Comment by th-otto** [comment]
No, please don't. AraNYM does not execute the mint loader, because there mint is loaded by the bootstrap feature and not from the auto folder.

**Comment by jflemaire** [comment]
> AraNYM does not execute the mint loader, because there mint is loaded by the bootstrap feature and not from the auto folder.
>
Only if the user uses of the bootstrap feature, which is not mandatory.

**Comment by th-otto** [comment]
But that would render that feature useless, and make it impossible to boot from a drive C which is a HOSTFS.

**Comment by skaftetryne** [comment]
On sø., 2017-05-07 at 00:20 -0700, Thorsten Otto wrote:
> But that would render that feature useless, and make it impossible to
> boot from a drive C which is a HOSTFS.

I don't think the limitations of a simulator should stand in the way for
useful functionality on actual hardware.

Jo Even
>

**Comment by th-otto** [comment]
What is useful in removing existing functionality? And the boostrap NatFeat is not a limitation, it's a feature that was designed for exactly this purpose. I don't think that existing funcionality should be dropped; i would call that a regression.

**Comment by mikrosk** [comment]
> But that would render that feature useless, and make it impossible to boot from a drive C which is a HOSTFS.
@th-otto wait, what? I totally don't understand. Aranym executes MINT.PRG with command line arguments like MEM_PROT=YES DEBUG_LEVEL=1. **It doesn't use *mint.ini* at all**. So why are you so worried about `mint.ini`?

**Comment by th-otto** [comment]
@mikrosk : because the same code that parses the ini file is also used to parse the commandline. If you remove it from mint, that won't work anymore. Also, if i remember right, the aros packet is initially set up in a way where mint + its configuration file is booted from a floppy.

**Comment by mikrosk** [comment]
@th-otto: ah, I see. I should have explained in more detail. Of course that the code parsing the command line *parameters* will stay! Only parsing of the ini *file* would be moved.

**Comment by th-otto** [comment]
But parsing the parameters from the ini file is the major part. Loading it to memory, then calling read_ini_file() is ~18 lines of code. I can't see any reason to remove that. Except maybe that after the change there is no reason to create it with default values.

**Comment by mikrosk** [comment]
@th-otto: no, the major part is the boot menu, this is what we're talking about (parsing ini is done mainly to initialise the boot menu).

I have nothing against keeping `read_ini_file()` in MINT.PRG, if nothing else it will be good for backward compatibility (same goes for default values). IMO only the boot menu should go.

**Comment by mikrosk** [comment]
Note for myself: maybe we even don't need another .ini format -- the loader would look into each `$SYSDIR` (enumerated from the list of MAJOR-MINOR-STATUS folders) and load the ini from there (and if not found, use default values) which would help keeping backward compatibility and allowing people ignore mint loader and/or its boot menu if they don't like it.
